datastore/src/main/java/org/jboss/pnc/datastore/repositories/audited/AuditedBuildConfigurationRepository.java

```diff
@@ -0,0 +1,23 @@
+package org.jboss.pnc.datastore.repositories.audited;
+
+import org.hibernate.envers.AuditReader;
+import org.jboss.pnc.datastore.audit.impl.AbstractAuditRepository;
+import org.jboss.pnc.model.BuildConfiguration;
+
+import javax.enterprise.context.ApplicationScoped;
+import javax.inject.Inject;
+
+@ApplicationScoped
+public class AuditedBuildConfigurationRepository extends AbstractAuditRepository<BuildConfiguration, Integer> {
+
+    @Deprecated
```

Comment:
CDI needs to have a default constructor for proxing. It shouldn't be 
used directly.

When we use @Deprecated annotation - your IDE will indicate that you 
shouldn't use it.

On 04/08/2015 04:12 PM, Paul Gier wrote:
>
> In 
> datastore/src/main/java/org/jboss/pnc/datastore/repositories/audited/AuditedBuildConfigurationRepository.java 
> <https://github.com/project-ncl/pnc/pull/278#discussion_r27972782>:
>
> > @@ -0,0 +1,23 @@
> > +package org.jboss.pnc.datastore.repositories.audited;
> > +
> > +import org.hibernate.envers.AuditReader;
> > +import org.jboss.pnc.datastore.audit.impl.AbstractAuditRepository;
> > +import org.jboss.pnc.model.BuildConfiguration;
> > +
> > +import javax.enterprise.context.ApplicationScoped;
> > +import javax.inject.Inject;
> > +
> > +@ApplicationScoped
> > +public class AuditedBuildConfigurationRepository extends AbstractAuditRepository<BuildConfiguration, Integer> {
> > +
> > +    @Deprecated
>
> Why is this marked deprecated? What should it be replaced by?
>
> —
> Reply to this email directly or view it on GitHub 
> <https://github.com/project-ncl/pnc/pull/278/files#r27972782>.
>